candig/server/backend.py

```diff
@@ -535,10 +592,65 @@ def countHelper(self, table, fv_counts):
             response_list.append(fv_counts)
         return {table: response_list}
 
+    def variantsResultsValidator(self, results):
+        """
+        Validates the results section
+        :param results: The results section of the request
+        :return: "variantsByGene" if the request contains gene, or "variants" if it contains start, end & referenceName
+        """
+        gene = results[0].get("gene")
+        start = results[0].get("start")
+        end = results[0].get("end")
+        referenceName = results[0].get("referenceName")
+
+        # Raise exception if gene is specified, but one or more of start, end or referenceName are also specified
+        if gene is not None and (start is not None or end is not None or referenceName is not None):
+            raise exceptions.MissingResultVariantKeysException
+
+        if gene is not None:
+            return "variantsByGene"
+
+        else:
+            # Raise exception if any of start, end, referenceName is None
+            if start is None or end is None or referenceName is None:
+                raise exceptions.MissingResultVariantKeysException
+
+            return "variants"
```

Comment:
I am not sure whether this is simpler or not:

`if gene:
    # Raise exception if gene is specified, but one or more of start,
    # end or referenceName are also specified
    if start or end or referenceName:
        raise exceptions.MissingResultVariantKeysException
    else:
        return "variantsByGene"

else:
    # Raise exception if any of start, end, referenceName is None
    if start and end and referenceName:
        return "variants"
    else:
        raise exceptions.MissingResultVariantKeysException
`

Reply:
Yes, good point. Too many `is not None`

Reply:
I also realized i specified the wrong exception classs...